pkg/monitor/pod.go

```diff
@@ -94,11 +95,21 @@ func startPodMonitoring(ctx context.Context, m Recorder, client kubernetes.Inter
 						Message: fmt.Sprintf("invariant violation (bug): static pod should not transition %s->%s with same UID", old, new),
 					})
 				default:
-					conditions = append(conditions, Condition{
-						Level:   Warning,
-						Locator: locatePod(pod),
-						Message: fmt.Sprintf("pod moved back to Pending"),
-					})
+					if old == corev1.PodRunning {
+						debugPod, _ := json.MarshalIndent(oldPod.Status, "", "  ")
+						conditions = append(conditions, Condition{
+							Level:   Warning,
+							Locator: locatePod(pod),
+							Message: fmt.Sprintf("pod moved back to Pending: old status: %s", string(debugPod)),
```

Comment:
This might be a little verbose but honestly I think it might not be the worst to merge this in and see if we find anything weird/unexpected? We should be able to glean any other info we need from the must-gather.

Reply:
I was actually opening this to get data from the jobs, not to merge.  Very likely to reproduce within 1-2 retests.